blurring_as_a_service/inference_pipeline/source/lock_file.py

```diff
@@ -0,0 +1,29 @@
+import os
+
+from blurring_as_a_service.utils.logging_handler import (  # noqa: E402
+    setup_azure_logging_from_config,
+)
+
+logger = setup_azure_logging_from_config()
+
+
+class LockFile:
+    def __init__(self, file_path):
+        self.file_path = file_path
+        self.locked_file_path = self.file_path + ".lock"
+
+    def __enter__(self):
+        try:
+            os.rename(self.file_path, self.locked_file_path)
+
+            with open(self.locked_file_path, "r") as src:
+                return src
+        except FileNotFoundError as e:
+            logger.info(f"File {self.locked_file_path} not found: {e}")
```

Comment:
I forgot here, let's raise the exceptions after logging them so who uses the object decides what to do with it.